Fluid/Filters/MiscFilters.cs

```diff
@@ -196,7 +196,7 @@ public static ValueTask<FluidValue> Base64UrlSafeEncode(FluidValue input, Filter
                 encodedBase64StringBuilder.Replace('+', '-');
                 encodedBase64StringBuilder.Replace('/', '_');
 
-                return new StringValue(encodedBase64StringBuilder.ToString());
+                return new StringValue(encodedBase64StringBuilder.ToString().TrimEnd('='));
```

Comment:
Does the string builder have something like it? That would prevent the extra string allocation.

Reply:
If we know that base64 encoded version ends with `==` (it should) it should suffice to `encodedBase64StringBuilder.Length -= 2;`

Reply:
That's a good option too

Reply:
It's not always the case, could be zero, one or two chars. See the test cases I shared for instance.

Reply:
But I see what you mean now, sorry.

Reply:
The issue the might be raise during decoding how we know there was zero, one or two "=" on the encoded URL?

Reply:
FYI there's `Base64.IsValid()` introduced in .NET 8.0 but it will not solve the problem in our case